README.md

```diff
@@ -1 +1,115 @@
+CCAO's iasWorld system-of-record. It is a replacement for
+[`service-sqoop-iasworld`](https://github.com/ccao-data/service-sqoop-iasworld),
+which is now deprecated.
+
+The Spark jobs pull iasWorld tables (or parts of tables) via
+[JDBC](https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html) and
+write them as [Hive-partitioned](https://duckdb.org/docs/data/partitioning/hive_partitioning.html)
+Parquet files to [AWS S3](https://aws.amazon.com/s3/). The Data Department then
+queries these Parquet files using [AWS Athena](https://aws.amazon.com/athena),
+giving us a 1-1 mirror of the system-of-record for analytical queries.
+
+## Submitting jobs
+
+> [!NOTE]
+> Before attempting to submit jobs to the cluster, first make sure the Spark
+> Docker Compose stack is active by running `docker compose up -d` in the
+> repository.
+
+`service-spark-iasworld` jobs are submitted via JSON, either as a string or
+as a file. All jobs should have the following format:
+
+```json
+{
+  "addn": {
+    "table_name": "iasworld.addn",
+    "min_year": 2020,
+    "max_year": 2024,
+    "cur": ["Y", "D"],
+    "predicates_path": "default_predicates.sql"
+  },
+  "job2": {
+    ...
+  }
+}
+```
+
+### Field definitions
+
+- `table_name (required)` - Name of the iasWorld table to extract, must be
+  prefixed with `iasworld.`.
+- `min_year (optional)` - Minimum tax year (inclusive) to extract from the
+  table. Defaults to `1999`.
+- `max_year (optional)` - Maximum tax year (inclusive) to extract from the
+  table. To extract a single year, set `min_year` and `max_year` to the same
+  value. Defaults to the current year.
+- `cur (optional)` - Values of the `cur` column to extract from the table.
+  Must be an array. Set to `null` in a job definition to ignore this column
+  when filtering and partitioning. Defaults to `["Y", "N", "D"]`.
+- `predicates_path (optional)` - String path to a SQL file within the `config/`
+  directory. The SQL file should define SQL BETWEEN expressions, where each
+  expression is one chunk that will be extracted by Spark during JDBC reads.
+  Expressions should not be overlapping. Set to `null` in a job definition
+  to disable using predicates completely. Defaults to `default_predicates.sql`.
+
+Long-lived job definitions are stored as YAML in `config/default_jobs.yaml`,
+then converted to JSON for submission. See `run.sh` for an example of this
+workflow using `yq`.
+
+### Predicates, filtering, and partitioning
+
+- **Predicates** are SQL statements used to chunk a table during reads
+  against the iasWorld database. The statements define mutually exclusive
+  queries that run in parallel (in order to speed up query execution).
+
+  Predicates are defined via a file of SQL statements in the `config/`
+  directory, then passed to each table job via a file path.
+- **Filters** are logic conditions applied to a query as part of the initial
+  read. They're effectively a SQL WHERE clause, and apply before each read.
```

Comment:
**[Question, non-blocking]** I originally thought "as part of the initial read" meant filters would be applied _during_ the read, but I think you mean they're applied _after_ the read, is that right? That is, filters are applied to the dataframe that results from the Spark read, whereas predicates are used to chunk the read itself?

Reply:
Filters are actually composed into each SQL query made to the DB using "predicate pushdown." That means they're executed as a part of each read, rather than after. I clarified this a bit in e41f8e608989a1ad373069bbfa2464a30dee0bd0.